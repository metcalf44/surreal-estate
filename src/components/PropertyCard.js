import React from "react";
import PropTypes from "prop-types";
import "../styles/PropertyCard.css";

const PropertyCard = ({
  title,
  type,
  city,
  bathrooms,
  bedrooms,
  price,
  email,
}) => {
  return (
    <div className="Property-card">
      <h2 className="card-heading" data-testid="card-heading">
        Surreal Estate
      </h2>
      <ul className="card-list">
        <li className="card-title">{title}</li>
        <li className="card-type">
          {type} - {city}
        </li>
        <li className="card-bedroom">{bedrooms} - bedrooms</li>
        <li className="card-bathroom">{bathrooms} - bathrooms </li>
        <li className="card-price">Â£{price}</li>
        <li className="card-email">
          <a href={`mailto:${email}`} className="card-mailto">
            Email seller
          </a>
        </li>
      </ul>
    </div>
  );
};

export default PropertyCard;

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  bedrooms: PropTypes.string.isRequired,
  bathrooms: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};
