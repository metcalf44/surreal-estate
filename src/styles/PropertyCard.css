.Property-card {
    background: rgb(171, 171, 249);
    display: inline-block;
    width: 45%;
    max-width: 300px;
    padding: .5em;
    margin: .5em;
}

ul {
    list-style-type: none;
}

.card-email {
    background: chocolate;
    margin: .25em 0;
    text-align: center;
    padding: .25em;
}
