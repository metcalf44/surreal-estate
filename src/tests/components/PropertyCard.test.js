import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "country cottage",
    type: "cottage",
    bathrooms: "2",
    bedrooms: "3",
    price: "250000",
    city: "Hitchin",
    email: "emetcalf44@gmail.com",
  };
  const { asFragment } = render(<PropertyCard {...validProps} />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values", () => {
    const { getByText } = render(<PropertyCard {...validProps} />);

    expect(getByText("cottage - Hitchin")).toHaveClass("card-type");
    expect(getByText("2 bathrooms")).toHaveClass("card-bathroom");
    expect(getByText("3 bedrooms")).toHaveClass("card-bedroom");
    expect(getByText("Â£250000")).toHaveClass("card-price");
    expect(getByText("email")).toHaveClass("card-email");
  });
});
