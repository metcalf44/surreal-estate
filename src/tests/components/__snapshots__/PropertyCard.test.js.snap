// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Property-card"
  >
    <h2
      class="card-heading"
      data-testid="card-heading"
    >
      Surreal Estate
    </h2>
    <ul
      class="card-list"
    >
      <li
        class="card-title"
      />
      <li
        class="card-type"
      >
         - 
      </li>
      <li
        class="card-bedroom"
      >
         - bedrooms
      </li>
      <li
        class="card-bathroom"
      >
         - bathrooms 
      </li>
      <li
        class="card-price"
      >
        Â£
      </li>
      <li
        class="card-email"
      >
        <a
          class="card-mailto"
          href="mailto:undefined"
        >
          Email seller
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
